### PR TITLE
Using olm catalog to deploy submariner

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

**Comment by skeeey** [comment]
we need wait for Submariner v0.8.0 release

**Comment by skeeey** [comment]
refer to https://github.com/open-cluster-management/backlog/issues/7536

File: deploy/config/crds/submariner.io_clusters_crd.yaml [hunk 1]
```diff
@@ -1,13 +1,63 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1alpha1
```
Comment (qiujian16) [inline deploy/config/crds/submariner.io_clusters_crd.yaml:1]: is this v1alpha1 or v1?

File: pkg/hub/submarineragent/controller.go [hunk 2]
```diff
@@ -302,3 +336,76 @@ func (c *submarinerAgentController) removeClusterRBACFiles(ctx context.Context,
...
+	clusterType, err := helpers.GetClusterType(clusterClient, config.ClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the cluster type %+v : %+v", config.ClusterName, err)
+	}
+	switch clusterType {
+	case helpers.ClusterTypeOCP:
+		//TODO: the scc should be handled in submariner v0.8.0, but need verify
+		config.NATEnabled = true
+		if version.IsDownStream() {
+			config.CatalogSource = "redhat-operators"
```
Comment (qiujian16) [inline pkg/hub/submarineragent/controller.go:405]: I  think we only  need to set different catalog source when it is in downstream. We may not set an explicit env to say it is downstream

File: pkg/hub/submarineragent/controller.go [hunk 3]
```diff
@@ -302,3 +336,76 @@ func (c *submarinerAgentController) removeClusterRBACFiles(ctx context.Context,
...
+		}
+		config.CatalogSourceNamespace = "openshift-marketplace"
```
Comment (qiujian16) [inline pkg/hub/submarineragent/controller.go:407]: i think this should also be set differently in downstream?

Comment (skeeey) [reply pkg/hub/submarineragent/controller.go:407]: it needn't, all catalogsources are in this namespace on ocp

File: pkg/hub/submarineragent/controller.go (hunk 2 above)
Comment (skeeey) [reply pkg/hub/submarineragent/controller.go:405]: good suggestion

File: deploy/config/crds/submariner.io_clusters_crd.yaml (hunk 1 above)
Comment (zhiweiyin318) [reply deploy/config/crds/submariner.io_clusters_crd.yaml:1]: yes, the CRD has been updated to apiextensions.k8s.io/v1.
suggest to get the CRD from submariner-operator repo by using `make` command.  cc @skeeey

Comment (skeeey) [reply deploy/config/crds/submariner.io_clusters_crd.yaml:1]: ok, I will

**Comment by zhiweiyin318** [comment]
@skeeey  `deploy/olm-catalog` and `csv` should be updated by `make update-csv` too.

**Comment by skeeey** [comment]
> @skeeey `deploy/olm-catalog` and `csv` should be updated by `make update-csv` too.

done

File: deploy/config/crds/submariner.io_clusters_crd.yaml (hunk 1 above)
Comment (skeeey) [reply deploy/config/crds/submariner.io_clusters_crd.yaml]: done

**Pull request merged**